src/validations/html/rules/MismatchedTag.js

```diff
@@ -0,0 +1,139 @@
+import {List, Map, Record, Set} from 'immutable';
+import Code from '../rules/Code';
+
+const Location = Record({row: 0, column: 0});  // eslint-disable-line new-cap
+
+export default class {
```

Comment:
> in that case I think it’s still better to show the two errors to get the students thinking more directly about what the correct HTML structure is.

Actually @outoftime, are two errors better even if the close tag proceeds the open tag?

```html
    <body>
        This is a paragraph.<p>

        Here is another one
    </body>
    </p>
```

![image](https://user-images.githubusercontent.com/775676/27570441-31021f5c-5acc-11e7-9e0e-e32083d03fa6.png)

Would these two errors be enough to let the average student realize that they misplaced the close tag? e.g. A `<p>` tag needs to be closed by a `</p>` tag, and I have a `</p>` that doesn't match any `<p>` tag; so I need to move that unmatched `</p>` to close the unclosed `<p>` tag.

If so, I'd be happy to get rid of the `misplaced-close-tag` handling, which would significantly reduce complexity.